internal/provider/function_compare_versions.go

```diff
@@ -64,6 +66,122 @@ func (f CompareVersionsFunction) Run(ctx context.Context, req function.RunReques
 	resp.Error = function.ConcatFuncErrors(resp.Result.Set(ctx, result))
 }
 
+func CompareVersionsRange(version, versionRanges, operator string) (bool, error) {
+	var result bool
+	var err error
+
+	if operator == "inside" || operator == "outside" {
+
+		comparisons, err := GetVersionComparisonsFromVersionRanges(versionRanges)
+		if err != nil {
+			return false, err
+		}
+
+		for _, comparison := range comparisons {
+
+			if comparison.Min == comparison.Max {
+
+				// When the min and max are the same, the version is singular which requires single comparison
+				result, err = CompareVersions(version, comparison.Min, "==")
+				if err != nil {
+					return false, err
+				}
+
+				// If the version is inside the range, the comparison must be true for result to be true
+				// If the version is outside the range, the comparison must be false for result to be true
+				if result {
+					result = true
```

Comment:
Why setting this if it is already true (your if condition)?

Reply:
not needed removed